test/es-module/test-esm-import-meta-main-eval.mjs

```diff
@@ -0,0 +1,77 @@
+import { spawnPromisified } from '../common/index.mjs';
+import * as fixtures from '../common/fixtures.js';
+import assert from 'node:assert/strict';
+
+const execOptions = ['--input-type', 'module'];
+
+const importMetaMainScript = `
+import assert from 'node:assert/strict';
+
+assert.strictEqual(import.meta.main, true, 'import.meta.main should evaluate true in main module');
+
+const { isMain: importedModuleIsMain } = await import(
+    ${JSON.stringify(fixtures.fileURL('es-modules/import-meta-main.mjs'))}
+);
+assert.strictEqual(importedModuleIsMain, false, 'import.meta.main should evaluate false in imported module');
+`;
+
+function wrapScriptInEvalWorker(script) {
+  return `
+  import { Worker } from 'node:worker_threads';
+  new Worker(${JSON.stringify(script)}, { eval: true });
+  `;
+}
+
+function convertScriptSourceToDataUrl(script) {
+  return new URL(`data:text/javascript,${encodeURIComponent(script)}`);
+}
+
+function wrapScriptInUrlWorker(script) {
+  return `
+  import { Worker } from 'node:worker_threads';
+  new Worker(new URL("${convertScriptSourceToDataUrl(script)}"));
```

Comment:
It's never safe to pass an arbitrary string as JS without wrapping in in `JSON.stringify`
```suggestion
  new Worker(new URL(${JSON.stringify(convertScriptSourceToDataUrl(script))}));
```

Reply:
Is that true even for URI-encoded content? I'm under the impression that all allowed URI characters are permitted without escaping in JS double-quoted strings

Reply:
You gain nothing from putting the quotes yourself, using `JSON.stringify` is the obvious choice if we follow the principle of least surprise.